src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -2164,5 +2164,407 @@ public class D { }
+        {
+            var source = @"
+class C
+{
+    bool M1(object o) => o is < (0.0d / 0.0d);
+    bool M2(object o) => o is < (0.0f / 0.0f);
+}";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview));
+            compilation.VerifyDiagnostics(
+                // (4,33): error CS8782: Relational patterns may not be used for a floating-point NaN.
+                //     bool M1(object o) => o is < (0.0d / 0.0d);
+                Diagnostic(ErrorCode.ERR_RelationalPatternWithNaN, "(0.0d / 0.0d)").WithLocation(4, 33),
+                // (5,33): error CS8782: Relational patterns may not be used for a floating-point NaN.
+                //     bool M2(object o) => o is < (0.0f / 0.0f);
+                Diagnostic(ErrorCode.ERR_RelationalPatternWithNaN, "(0.0f / 0.0f)").WithLocation(5, 33));
+        }
+
+        [Fact]
+        public void Relational_07()
+        {
+            var source = @"
+class C
+{
+    public bool M(char c) => c switch
+    {
+        >= 'A' and <= 'Z' or >= 'a' and <= 'z' => true,
+        'a'                                    => true, // error 1
+        > 'k' and < 'o'                        => true, // error 2
+        '0'                                    => true,
+        >= '0' and <= '9'                      => true,
+        _                                      => false,
+    };
+}";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview));
+            compilation.VerifyDiagnostics(
+                // (7,9): error CS8510: The pattern has already been handled by a previous arm of the switch expression.
+                //         'a'                                    => true, // error 1
+                Diagnostic(ErrorCode.ERR_SwitchArmSubsumed, "'a'").WithLocation(7, 9),
+                // (8,9): error CS8510: The pattern has already been handled by a previous arm of the switch expression.
+                //         > 'k' and < 'o'                        => true, // error 2
+                Diagnostic(ErrorCode.ERR_SwitchArmSubsumed, "> 'k' and < 'o'").WithLocation(8, 9)
+                );
+        }
+
+        [Fact]
+        public void Relational_08()
+        {
+            var source = @"
+class C
+{
+    public int M(uint c) => c switch
+    {
+        >= 5 => 1,
+        4 => 2,
+        3 => 3,
+        2 => 4,
+        1 => 5,
+        0 => 6,
+        _ => 7,
+    };
+}";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview));
+            compilation.VerifyDiagnostics(
+                // (12,9): error CS8510: The pattern has already been handled by a previous arm of the switch expression.
+                //         _ => 7,
+                Diagnostic(ErrorCode.ERR_SwitchArmSubsumed, "_").WithLocation(12, 9)
+                );
+        }
```

Comment:
>} [](start = 8, length = 1)

suggestion: Do we have a test for `is not object`? Is that always false?

Reply:
suggestion: Can we add some tests with enums of different types too?

---
In reply to: [381013785](https://github.com/dotnet/roslyn/pull/41503#discussion_r381013785) [](ancestors = 381013785)

Reply:
`is not object` has the same effect as `is null`.  Added an `is not object` test in `NullableReferenceTypesVsPatterns.NotNullIsAPureNullTest` to demonstrate.

Added signed and unsigned enum tests in `PatternMatchingTests3`.


---
In reply to: [381448840](https://github.com/dotnet/roslyn/pull/41503#discussion_r381448840) [](ancestors = 381448840,381013785)